### DISCUSSION TITLE
Cross Validation Hold-out & K-Fold

### DISCUSSION THREAD

**Discussion by dev-fredericfox** [BrainJS/brain.js#748] [Q&A] [answered]
The documentation doesn't really state what type of Cross Validation brain is doing. But since there is an optional K value, I assume it's a K-Fold. What is the default value if K is omitted? Or is it determined based on the dataset?

Is there a way to train on 80% of the data and cross validate with remaining 20% (sometimes called Hold-out)?

**Comment by robertleeplummerjr** [comment, marked as answer]
>The documentation doesn't really state what type of Cross Validation brain is doing. But since there is an optional K value

That is correct, it is K-Fold

>What is the default value if K is omitted?:

4 as stated here - https://github.com/BrainJS/brain.js/blob/5d523fe241f5edddd220e19420dfcecab841b30c/src/cross-validate.ts#L184

>Is there a way to train on 80% of the data and cross validate with remaining 20% (sometimes called Hold-out)?

I don't believe so.  The data is essentially split into k folds, and trained on, and then later compared.
